site/gatsby-site/src/utils/updateTsne.js

```diff
@@ -0,0 +1,63 @@
+const TSNE = require('tsne-js');
+
+const config = require('../../config');
+
+const { MongoClient } = require('mongodb');
+
+const BSON = require('bson');
+
+const updateTsneInDatabase = async () => {
+  const mongo = new MongoClient(config.mongodb.translationsConnectionString);
+
+  await mongo.connect();
+
+  const incidentsCollection = mongo.db(config.realm.production_db.db_name).collection('incidents');
+
+  const incidentsWithEmbeddings = await incidentsCollection
+    .find({ embedding: { $exists: true } })
+    .toArray();
+
+  const embeddings = incidentsWithEmbeddings.map((incident) => incident?.embedding?.vector);
+
+  const ids = incidentsWithEmbeddings.map((incident) => incident.incident_id);
+
+  const model = new TSNE({
+    dim: 2,
+    perplexity: 30.0,
+    earlyExaggeration: 3.0,
+    learningRate: 100.0,
+    nIter: 10000,
+    metric: 'euclidean',
+  });
+
+  // inputData is a nested array which can be converted into an ndarray
+  // alternatively, it can be an array of coordinates (second argument should be specified as 'sparse')
+  model.init({ data: embeddings, type: 'dense' });
+
+  // `error`,  `iter`: final error and iteration number
+  // note: computation-heavy action happens here
+  const [err, iter] = model.run();
+
+  if (err) {
```

Comment:
Not sure about this, should the build continue if there is an error here?

Like, do we want to still deploy the site if this failed?

Reply:
It seems like there's always an error in one or two of the 10000 iterations. I'm not sure what an "error" actually means here, but it doesn't seem to make any difference.